### PR TITLE
Add config options, default configuration and support for Azure OpenAI

### LINKED ISSUE #21: Option to configure API key for instance
For my use-case I'd like to host chatpad in a private space for several users who will have access.
Current implementation is great for privacy and shared deployments for different users, but it's a bit hard to explain non-technical users how to insert their key and configure model for usage.

It would be great if there would be an option to configure an API key which will be used for all users.
For the docker deployment(hoping https://github.com/deiucanta/chatpad/pull/8 is getting merged soon :) ) it would be great to have option to define an env variable which will be used for this.

Since this is a static front-end applications I can suggest 2 options to implement this atm:
- to use some bash scripting for docker container startup and embed key value in some special placeholder.
  This option would require to define some variable in code and set a unique value which could be replaced when container will start. While I've used similar approach in the past, and it worked, but it seems to be very hacky way for this.
- to write a key into some file and read it with regular HTTP request served by nginx. Application than can just request key and use it.

Option 2 seems to be less secure than option one, but both options will have a key in plaintext when just browsing source. For my use-case it is safe to have key in plaintext without any authentication since it will only be reachable inside my network, so there is no risk of leaking key to public.

I will be happy to help with scripting/docker setup if this will be implemented and any help will be needed.

### PR DESCRIPTION
**Pull request by itbm**
Added config options, default configuration via a `config.json` file, plus support for custom endpoints including Azure OpenAI.

Closes #21 and #16

### CODE REVIEW

**Comment by wescrockett** [comment]
I currently have my chatbot accessible through Cloudflare tunnels with authentication. On a network that blocks *.openai.com, when I enter my API key, it gets blocked when validating. I assume adding it to the config file would bypass the initial check... at that point, does the server hosting Chatbot handle all communications with openai.com or is the client browser still making calls to openai?

**Comment by Supernova3339** [comment]
If you are blocking the API endpoint, I would advise you to unblock it to
ensure a proper connection. Any issues that arise from blocking the
endpoint are yours to solve.

**Comment by wescrockett** [comment]
Ah, yeah, i understand that for sure. Let me rephrase though (meant it as a general question):

Other than validating the API key, does the server running the interface act as a proxy for the API or are the API calls still made client side? I am running the interface on a home server and accessing the server from a network that has openai.com blocked. My *hope* was that all requests/responses to/from api.openai.com would be handled server side. If that isn't how this is made, that's fine, just trying to make sure I understand it.

Thank you.

**Comment by zekker6** [comment]
@wescrockett This is a client-side application, so all requests will be executed client-side. Unfortunately, In your case that will mean that requests will be blocked.

**Comment by wescrockett** [comment]
> This is a client-side application, so all requests will be executed client-side. Unfortunately, In your case that will mean that requests will be blocked.

Thank you for the clarification.

**Comment by kienerj** [comment]
has this been merged? Alternative end-point to openai would be very helpful to have.

**Comment by Supernova3339** [comment]
No, would be a breaking change with the current merges, I'd need to redo
the PR myself.
